backend-config/namespace_config.go

```diff
@@ -112,14 +124,21 @@ func (nc *namespaceConfig) getFromAPI(ctx context.Context) (map[string]ConfigT,
 		return config, err
 	}
 
+	if !useUpdateAfter {
+		nc.workspacesConfig = make(map[string]ConfigT)
+	}
+
 	for workspaceID, wc := range workspacesConfig {
 		// always set connection flags to true for hosted and multi-tenant warehouse service
 		wc.ConnectionFlags.URL = nc.cpRouterURL
 		wc.ConnectionFlags.Services = map[string]bool{"warehouse": true}
-		workspacesConfig[workspaceID] = wc
+		nc.workspacesConfig[workspaceID] = wc
```

Comment:
One question. What if one of the incremental updates something gets deleted? I see that you only add to the map but never delete. I was expecting something like another loop that iterates over `nc.workspacesConfig` and deletes the entry if not present in the new `workspacesConfig`. Would control plane send us such updates?

Reply:
Yeah, thats an interesting question. Let me check with them about this

Reply:
I just had a chat with @fxenik and @ktgowtham and we agreed on the following structure:

![Screenshot from 2023-04-24 09-56-02](https://user-images.githubusercontent.com/1288256/233935781-169c1aeb-9526-4a49-bf53-999caa8f3d58.png)

This means that workspaces with `null` will have to be skipped since they don't have any relevant changes for us. Then we'll loop over `nc.workspacesConfig` instead and if we don't have a workspaceID at all in the received response we delete it from `nc.workspacesConfig`.

**NOTE:** This PR should be on hold until they are done with the aforementioned changes.